pkg/utils/deployer.go

```diff
@@ -444,38 +453,43 @@ func applyResourceWithRetry(ctx context.Context, dynamicClient dynamic.Interface
 			return false, nil
 		}
 
-		// try  to update resource
-		_, lastError = dynamicClient.Resource(restMapping.Resource).Namespace(resource.GetNamespace()).
-			Update(context.TODO(), resource, metav1.UpdateOptions{})
-		if lastError == nil {
-			return true, nil
-		}
-		statusCauses, ok := getStatusCause(lastError)
-		if !ok {
-			lastError = fmt.Errorf("failed to get StatusCause for %s %s", resource.GetKind(), klog.KObj(resource))
-			return false, nil
-		}
-
 		curObj, err := dynamicClient.Resource(restMapping.Resource).Namespace(resource.GetNamespace()).
 			Get(context.TODO(), resource.GetName(), metav1.GetOptions{})
 		if err != nil {
 			lastError = err
 			return false, nil
-		}
-		resourceCopy := resource.DeepCopy()
-		for _, cause := range statusCauses {
-			if cause.Type != metav1.CauseTypeFieldValueInvalid {
-				continue
+		} else {
+			lastError = nil
+		}
+		rawSpec, _, _ := unstructured.NestedMap(resource.Object, "spec")
```

Comment:
Same. `spec` is not the only part we need to compare. We need to have a more generic way.